Panic when configuring `[Storage.S3]`

**Describe the bug**
When we configure our instance with:

```console
        [Storage.S3]
            Region = "us-east-1"
            Bucket = "my-s3-bucket-name"
            CredentialsEndpoint = "http://169.254.169.254"
```

we get a panic as follows:

**Error Message**

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x6a39b2]

goroutine 145 [running]:
net/http.(*Client).deadline(0x0, 0x153d740, 0x203001, 0xc0000db030)
	/usr/local/go/src/net/http/client.go:189 +0x22
net/http.(*Client).do(0x0, 0xc0006a4400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:585 +0x2ae
net/http.(*Client).Do(...)
	/usr/local/go/src/net/http/client.go:567
github.com/aws/aws-sdk-go/aws/corehandlers.sendFollowRedirects(0xc0006ae500, 0xc0006a4400, 0x145cc40, 0x0)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/corehandlers/handlers.go:120 +0x3b
github.com/aws/aws-sdk-go/aws/corehandlers.glob..func3(0xc0006ae500)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/corehandlers/handlers.go:112 +0xa1
github.com/aws/aws-sdk-go/aws/request.(*HandlerList).Run(0xc0006ae708, 0xc0006ae500)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/request/handlers.go:267 +0x98
github.com/aws/aws-sdk-go/aws/request.(*Request).sendRequest(0xc0006ae500, 0x0, 0x0)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/request/request.go:578 +0xb4
github.com/aws/aws-sdk-go/aws/request.(*Request).Send(0xc0006ae500, 0x0, 0x0)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/request/request.go:531 +0x107
github.com/aws/aws-sdk-go/aws/credentials/endpointcreds.(*Provider).getCredentials(0xc0002b3360, 0x1b67aa0, 0xc000042028, 0xc00034c000, 0x7fae92a98fe8, 0x80010000000000)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/credentials/endpointcreds/provider.go:171 +0x319
github.com/aws/aws-sdk-go/aws/credentials/endpointcreds.(*Provider).RetrieveWithContext(0xc0002b3360, 0x1b67aa0, 0xc000042028, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/credentials/endpointcreds/provider.go:125 +0x7d
github.com/aws/aws-sdk-go/aws/credentials/endpointcreds.(*Provider).Retrieve(0xc0002b3360, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x29, ...)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/credentials/endpointcreds/provider.go:119 +0x96
github.com/aws/aws-sdk-go/aws/credentials.(*ChainProvider).Retrieve(0xc000436c60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/credentials/chain_provider.go:75 +0x102
github.com/aws/aws-sdk-go/aws/credentials.(*Credentials).singleRetrieve(0xc000436c90, 0x1b67ce0, 0xc0000db020, 0x4, 0x0, 0x0, 0xc0002ec790)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/credentials/credentials.go:263 +0x3f6
github.com/aws/aws-sdk-go/aws/credentials.(*Credentials).GetWithContext.func1(0x16, 0x0, 0x0, 0x0)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/aws/credentials/credentials.go:244 +0x8a
github.com/aws/aws-sdk-go/internal/sync/singleflight.(*Group).doCall(0xc000436ca0, 0xc0006bc180, 0x0, 0x0, 0xc0006ac140)
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/internal/sync/singleflight/singleflight.go:97 +0x2e
created by github.com/aws/aws-sdk-go/internal/sync/singleflight.(*Group).DoChan
	/go/src/github.com/gomods/athens/vendor/github.com/aws/aws-sdk-go/internal/sync/singleflight/singleflight.go:90 +0x2b4
```

**To Reproduce**
I'm not sure what is causing this; it's possible that the kube2iam isn't propagating, for example, or that the bucket permissions aren't set, or that we're missing something for the URL but none of these sorts of predictable failure conditions should be able to cause a panic.

**Expected behavior**
We get an error message that explains what we did wrong

**Environment (please complete the following information):**
 - OS: Linux 64bit
 - Go version : 1.17
 - Proxy version : v0.11.0
 - Storage (fs/mongodb/s3 etc.) : s3

**Suspicions**

I suspect that

https://github.com/gomods/athens/blob/0c78ebd6308329713a2d2dd9bc007ca9c36d98cc/pkg/storage/s3/s3.go#L38

should be calling

https://github.com/aws/aws-sdk-go/blob/5806617529e1ded4f995062aa38c915727ed912b/aws/defaults/defaults.go#L55

--- a/pkg/storage/s3/s3.go
+++ b/pkg/storage/s3/s3.go
@@ -35,7 +35,7 @@ type Storage struct {
 func New(s3Conf *config.S3Config, timeout time.Duration, options ...func(*aws.Config)) (*Storage, error) {
 	const op errors.Op = "s3.New"
 
-	awsConfig := &aws.Config{}
+	awsConfig := defaults.Config()
 	awsConfig.Region = aws.String(s3Conf.Region)
 	for _, o := range options {
 		o(awsConfig)

